src/Components/Search/NewSearch/NewSearchInputPillsContainer.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  Box,
+  Flex,
+  Pill,
+  ShelfNavigationProps,
+  ShelfNext,
+  ShelfPrevious,
+} from "@artsy/palette"
+import { useEffect, useRef, useState } from "react"
+import styled, { css } from "styled-components"
+
+const PILLS_OPTIONS = [
+  "TOP",
+  "ARTWORK",
+  "ARTIST",
+  "ARTICLE",
+  "SALE",
+  "ARTIST_SERIES",
+  "COLLECTION",
+  "FAIR",
+  "SHOW",
+  "GALLERY",
+]
+
+const ChevronStyle = css`
+  position: absolute;
+  width: 30px;
+  height: 30px;
+  @media (hover: none) {
+    display: none;
+  }
+`
+
+const PreviousChevron = styled(ShelfPrevious)<ShelfNavigationProps>`
+  ${ChevronStyle}
+`
+
+const NextChevron = styled(ShelfNext)<ShelfNavigationProps>`
+  ${ChevronStyle}
+`
+
+interface GradientBgProps {
+  placement: "right" | "left"
+}
+
+const GradientBg = styled(Box)<GradientBgProps>`
+  position: "absolute";
+  right: 0;
+  width: 100px;
+  height: 30px;
+  background: transparent;
+  background-image: linear-gradient(
+    to ${props => props.placement},
+    rgba(255, 255, 255, 0) 0%,
+    rgba(255, 255, 255, 1) 40%
+  );
+`
+
+const PillsContainer = styled(Flex)`
+  flex-wrap: nowrap;
+  overflow-x: auto;
+  scrollbar-width: none;
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+`
+
+const GRADIENT_BG_WIDTH = 30
+
+export const NewSearchInputPillsContainer = () => {
+  const pillsRef = useRef<HTMLDivElement>(null)
+  const [showPreviousChevron, setShowPreviousChevron] = useState<boolean>(false)
+  const [showNextChevron, setShowNextChevron] = useState<boolean>(false)
+
+  // On inital load
+  useEffect(() => {
+    if (pillsRef.current) {
+      const pillsContainer = pillsRef.current
+      setShowNextChevron(
+        pillsContainer.scrollWidth > pillsContainer.clientWidth
+      )
+    }
+  }, [])
+
+  // when screen width changes
+  useEffect(() => {
+    const handleResize = () => {
+      if (pillsRef.current) {
+        const pillsContainer = pillsRef.current
+        if (pillsContainer.scrollWidth <= pillsContainer.clientWidth) {
+          setShowNextChevron(false)
+        } else {
+          setShowNextChevron(true)
+        }
+      }
+    }
+
+    window.addEventListener("resize", handleResize)
+    return () => window.removeEventListener("resize", handleResize)
+  }, [])
+
+  const scrollToLeft = () => {
+    if (pillsRef.current) {
+      const pillsContainer = pillsRef.current
+      const pills = pillsContainer.children
+      const currentScroll = pillsContainer.scrollLeft
+      const visibleWidth = pillsContainer.offsetWidth
+      let scrollBy = 0
+
+      // Find the last visible pill and the next pill
+      for (let i = pills.length - 1; i >= 0; i--) {
+        const pill = pills[i] as HTMLElement
+        if (
+          pill.offsetLeft + pill.offsetWidth <=
+          currentScroll + visibleWidth
+        ) {
+          const prevPill = pills[i - 1] as HTMLElement
+          scrollBy = -prevPill.offsetWidth
+          break
+        }
+      }
```

Comment:
**suggestion**: i think we can do this logic once when mounting the component or when receiving the `resize` event

Reply:
how do we do it for [Carousel](https://github.com/artsy/palette/blob/main/packages/palette/src/elements/Carousel/Carousel.tsx#L133-L144)

Reply:
Ah, maybe I misunderstood

Reply:
**question**: can we use the same approach with `useCursor` hook, as we do with the `Carousel` ([link 1](https://github.com/artsy/palette/blob/main/packages/palette/src/elements/Carousel/Carousel.tsx#L133-L144), [link 2](https://github.com/artsy/palette/blob/main/packages/palette/src/elements/Carousel/Carousel.tsx#L128)) and `Shelf` ([link 1](https://github.com/artsy/palette/blob/main/packages/palette/src/elements/Shelf/Shelf.tsx#L117), [link 2](https://github.com/artsy/palette/blob/main/packages/palette/src/elements/Shelf/Shelf.tsx#L156)) components?

Reply:
or better
1. get pills offsets on mount/resize event and keep it in local state (e.g. `offsets = [0, 100, 150, 250, 300]`)
2. keep index of current offset in local state
4. when user clicks the right chevron button -> scroll to the next offset (e.g. `scrollTo(offsets[currentOffsetIndex + 1])`)
5. when user clicks the left chevron button -> scroll to the prev offset (e.g. `scrollTo(offsets[currentOffsetIndex - 1])`)
6. when user scrolls - get nearest offset index and save it in local state (e.g. [like this](https://github.com/artsy/palette/blob/ffa2c198375e7853516cae20d7f6002db48b4b5c/packages/palette/src/elements/Shelf/Shelf.tsx#L110-L117))

You can find more examples/code in Shelf component: https://github.com/artsy/palette/blob/ffa2c198375e7853516cae20d7f6002db48b4b5c/packages/palette/src/elements/Shelf/Shelf.tsx#L133-L166

Reply:
Pushed some code with the proposed changes to the [dimatretyak/sandbox/scrollable-pills](https://github.com/artsy/force/compare/dimatretyak/sandbox/scrollable-pills) branch. This is not the final version yet and improvements are needed, but it is already close to what we want (as it seems to me)


https://user-images.githubusercontent.com/3513494/233492367-730580d3-d3f6-4aea-82c6-b10dbd64a52f.mp4

Reply:
Thanks for the suggestions, Dima. I see what you mean. 
My use case is when the user clicks the next button the next pill should be fully visible. I know you are noticing the pill next to the previous button. Which is also good to take into account. But it would make more sense if the pill near the next button is visible when the next button is clicked and the pill near the previous button is visible when the previous button is clicked. 
What do you think?

Reply:
I would love to pair on this with you :)